firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistributionApi.java

```diff
@@ -0,0 +1,137 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.appdistribution;
+
+import androidx.annotation.NonNull;
+import com.google.android.gms.tasks.Task;
+import java.util.Optional;
+import org.jetbrains.annotations.Nullable;
+
+public abstract class FirebaseAppDistributionApi {
+
+  /**
+   * Updates the app to the latest release, if one is available. Returns the release information or
+   * an empty Optional if no update is found. Performs the following actions: 1. If tester is not
+   * signed in, presents the tester with a Google sign in UI 2. Checks if a newer release is
+   * available. If so, presents the tester with a confirmation dialog to begin the download. 3. For
+   * APKs, downloads the binary and starts an installation intent. 4. For AABs, directs the tester
+   * to the Play app to complete the download and installation.
+   */
+  @NonNull
+  public abstract Task<Optional<AppDistributionRelease>> updateToLatestRelease();
+
+  /**
+   * Returns an AppDistributionRelease if one is available for the current signed in tester. If no
+   * update is found, returns an empty Optional. If tester is not signed in, presents the tester
+   * with a Google sign in UI
+   */
+  @NonNull
+  public abstract Task<Optional<AppDistributionRelease>> checkForUpdate();
+
+  /**
+   * Updates app to the latest release. If the latest release is an APK, downloads the binary and
+   * starts an installation If the latest release is an AAB, directs the tester to the Play app to
+   * complete the download and installation.
+   *
+   * @throws an UPDATE_NOT_AVAIALBLE exception if
+   * @param updateProgressListener a callback function invoked as the update progresses
+   */
+  @NonNull
+  public abstract Task<Void> updateApp(@Nullable UpdateProgressListener updateProgressListener);
```

Comment:
it's a bit strange to me that lack of update is an exception, do you have an api review doc I can take a look at?

Reply:
This method is different from updateToLatestRelease, because it should be called only after `checkForUpdate` returns a truthy value. If it is called when no release is available, we'd throw the exception.

We'll share the API proposal with you over chat